stk/calculators/optimization/optimizers.py

```diff
@@ -773,19 +1178,18 @@ def optimize(self, mol, conformer=-1):
 
         os.mkdir(output_dir)
         init_dir = os.getcwd()
+        os.chdir(output_dir)
+
         try:
-            os.chdir(output_dir)
-            xyz = join(output_dir, f'input_structure.xyz')
-            mol.write(xyz, conformer=conformer)
-            cmd = [
-                self.gfnxtb_path,
-                xyz,
-                '-opt',
-                '--parallel',
-                self.num_cores
-            ]
-            sp.run(cmd, stdin=sp.PIPE, stdout=sp.PIPE, stderr=sp.PIPE)
-            output_xyz = join(output_dir, 'xtbopt.xyz')
-            mol.update_from_xyz(path=output_xyz, conformer=conformer)
+            result = self._run_optimizations(mol, conformer)
         finally:
             os.chdir(init_dir)
+
+        if result is None:
+            self.incomplete.add((mol, conformer))
+            logging.warning(
+                'Negative frequencies present in '
+                f'{mol.name} conformer {conformer} '
+                f'after {self.max_runs} optimizations '
+                'completed.'
+            )
```

Comment:
In this function, because (mol, conformer) is used so frequently, i would reccomend placing it into a variable.

Reply:
The error message will have to be changed however, because the function can exit early, ie when runs performed is less than max_runs and still return False.

Reply:
Maybe changing it to something along the lines of "negative frequencies present in final structure of {mol.name} conformer {conformer}"

Reply:
If you really want to make the user know how many runs were done, you can always done always use the logger (maybe info level?) if you think its necessary / beneficial

Reply:
please check the logic I have used in this funciton and the one above